lxd/instance/drivers/driver_qemu.go

```diff
@@ -5777,12 +5777,18 @@ func (d *qemu) writeInstanceData() error {
 		location = d.Location()
 	}
 
+	id := d.LocalConfig()["volatile.cloud-init.instance-id"]
```

Comment:
This logic is duplicated in devlxdMetadataGet. If it ever changes we would need to remember to find and change it in 2 places. Instead, would it make sense to add this to the instance.Instance interface as `CloudInitID()` and then put this logic in the instance common driver?